Mathlib/Algebra/Quaternion.lean

```diff
@@ -639,55 +654,63 @@ theorem im_star : (star a).im = -a.im :=
   QuaternionAlgebra.ext neg_zero.symm rfl rfl rfl
 
 @[simp]
-theorem star_mk (a₁ a₂ a₃ a₄ : R) : star (mk a₁ a₂ a₃ a₄ : ℍ[R,c₁,c₂]) = ⟨a₁, -a₂, -a₃, -a₄⟩ :=
-  rfl
+theorem star_mk (a₁ a₂ a₃ a₄ : R) : star (mk a₁ a₂ a₃ a₄ : ℍ[R,c₁,c₂,c₃]) =
+    ⟨a₁ + c₂ * a₂, -a₂, -a₃, -a₄⟩ := rfl
 
-instance instStarRing : StarRing ℍ[R,c₁,c₂] where
+instance instStarRing : StarRing ℍ[R,c₁,c₂,c₃] where
   star_involutive x := by simp [Star.star]
-  star_add a b := by ext <;> simp [add_comm]
+  star_add a b := by ext <;> simp [add_comm] ; ring
   star_mul a b := by ext <;> simp <;> ring
 
-theorem self_add_star' : a + star a = ↑(2 * a.re) := by ext <;> simp [two_mul]
+theorem self_add_star' : a + star a = ↑(2 * a.re + c₂ * a.imI) := by ext <;> simp [two_mul]; ring
 
-theorem self_add_star : a + star a = 2 * a.re := by simp only [self_add_star', two_mul, coe_add]
+theorem self_add_star : a + star a = 2 * a.re + c₂ * a.imI := by
+  simp [self_add_star']; rfl
```

Comment:
This `rfl` should not be needed. The missing lemma (which can go on line 501 or so, just after `coe_mul` is 

```
@[norm_cast, simp]
lemma coe_ofNat {n : ℕ} [n.AtLeastTwo]: ((ofNat(n) : R) :
    ℍ[R,c₁,c₂,c₃]) = (ofNat(n) : ℍ[R,c₁,c₂,c₃]) := by
  rfl
```